core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -345,6 +391,9 @@ private static class DefaultClientRequestMetrics
             final String timeouts = idPrefix.name("timeouts");
             writeTimeouts = parent.counter(timeouts, idPrefix.tags("cause", "WriteTimeoutException"));
             responseTimeouts = parent.counter(timeouts, idPrefix.tags("cause", "ResponseTimeoutException"));
+
+            failureAttemptsWithCause = new HashMap<>();
+            failureAttemptsWithHttpStatus = new HashMap<>();
```

Comment:
To be honest, I don't think this optimization is really necessary since we'll be doing a map search once anyways when calling `MeterRegistry#counter`.